spec/components/metrics_summary_component/view_spec.rb

```diff
@@ -63,19 +75,45 @@
     end
   end
 
-  context "when metrics_data has a value for weekly submissions" do
-    let(:metrics_data) { { weekly_submissions: 1235, form_is_new: false } }
+  context "when weekly starts is zero" do
+    let(:metrics_data) { { weekly_submissions: 0, form_is_new: false, weekly_starts: 0 } }
+
+    it "returns the 'no starts' message" do
+      expect(metrics_summary.error_message).to eq(I18n.t("metrics_summary.errors.no_submissions_html"))
+    end
+
+    it "renders the error message" do
+      expect(page).to have_text(Nokogiri::HTML(metrics_summary.error_message).text)
```

Comment:
Why are we using Nokogiri here?

Reply:
As far as I can tell Capybara doesn't have matcher for checking whether the page contains a specific bit of HTML (these are HTML translations). So the best solution I could think of was to use the `has_text` matcher and use Nokogiri to get the translation's text value.

Reply:
🤔  you should be able to `find` html using selectors but this specific test looks like it just looking for specific text on the page rather than specific HTML with text.

Reply:
I think the finders would be on the `expect` side of the assertion (and I should probably add these). My issue is that I don't think Capybara provides a matcher to compare whatever's in the `expect` to the translation.

So I can do:

```ruby
  expect(page).to have_text(Nokogiri::HTML(metrics_summary.error_message).text)
```
or: 
```ruby
  expect(page.find(".govuk-inset-text")).to have_text(Nokogiri::HTML(metrics_summary.error_message).text)
```
or:
```ruby
  expect(page).to have_css(".govuk-inset-text", text: Nokogiri::HTML(metrics_summary.error_message).text)
```

but not:

```ruby
  expect(page).to have_text(metrics_summary.error_message)
```
or: 
```ruby
  expect(page.find(".govuk-inset-text")).to have_text(metrics_summary.error_message)
```
or:
```ruby
  expect(page).to have_css(".govuk-inset-text", text: metrics_summary.error_message)
```
because in this case `error_message` is a HTML string `"<p>Sorry, there's a problem getting your form's metrics.</p><p>Try again later.</p>"` and Capybara's matchers will strip the HTML tags from whatever's inside the `expect()`. So unless I'm missing something I think I have to either strip out the HTML tags on the right hand side as I'm doing here (either with Nokogiri or something else), or I'll have to hard-code text into the tests to get around the HTML tag stripping:
```ruby
  expect(page.find(".govuk-inset-text > p")).to have_text("Sorry, there's a problem getting your form's metrics.")
  expect(page.find(".govuk-inset-text > p")).to have_text("Try again later.")
```

Reply:
Ok, not to worry lets leave it as is

Reply:
I'll change them to `expect(page).to have_css(".govuk-inset-text", text: metrics_summary.error_message)` now, since I think that's a better, more specific test. Thanks for raising this!